src/main/java/org/disq_bio/disq/impl/file/FileSystemWrapper.java

```diff
@@ -39,6 +40,49 @@
  */
 public interface FileSystemWrapper extends Serializable {
 
+  /** Represents a file in a directory listing. */
+  class FileStatus implements Comparable<FileStatus> {
+    private final String path;
+    private final long length;
+
+    public FileStatus(String path, long length) {
+      this.path = path;
+      this.length = length;
+    }
+
+    /** @return the file path */
+    public String getPath() {
+      return path;
+    }
+
+    /** @return the length of the file in bytes */
+    public long getLength() {
+      return length;
+    }
+
+    @Override
+    public int compareTo(FileStatus o) {
+      int cmp = path.compareTo(o.path);
```

Comment:
`compareTo` should sort nulls first or last to prevent NPE

Reply:
Fixed